extensions/guacamole-auth-json/src/main/java/org/apache/guacamole/auth/json/CryptoService.java

```diff
@@ -0,0 +1,215 @@
+     * used by decrypt().
+     *
+     * @param keyBytes
+     *     The raw bytes from which the encryption/decryption key should be
+     *     generated.
+     *
+     * @return
+     *     A new key suitable for encryption or decryption, generated from the
+     *     given bytes.
+     */
+    public SecretKey createEncryptionKey(byte[] keyBytes) {
+        return new SecretKeySpec(keyBytes, DECRYPTION_KEY_GENERATION_ALGORITHM_NAME);
+    }
+
+    /**
+     * Creates a new key suitable for signature verification using the provided
+     * raw key bytes. The algorithm used to generate this key is dictated by
+     * SIGNATURE_KEY_GENERATION_ALGORITHM_NAME and must match the algorithm
+     * used by sign().
+     *
+     * @param keyBytes
+     *     The raw bytes from which the signature verification key should be
+     *     generated.
+     *
+     * @return
+     *     A new key suitable for signature verification, generated from the
+     *     given bytes.
+     */
+    public SecretKey createSignatureKey(byte[] keyBytes) {
+        return new SecretKeySpec(keyBytes, SIGNATURE_KEY_GENERATION_ALGORITHM_NAME);
+    }
+
+    /**
+     * Decrypts the given ciphertext using the provided key, returning the
+     * resulting plaintext. If any error occurs during decryption at all, a
+     * GuacamoleException is thrown. The IV used for the decryption process is
+     * a null IV (all binary zeroes).
+     *
+     * @param key
+     *     The key to use to decrypt the provided ciphertext.
+     *
+     * @param cipherText
+     *     The ciphertext to decrypt.
+     *
+     * @return
+     *     The plaintext which results from decrypting the ciphertext with the
+     *     provided key.
+     *
+     * @throws GuacamoleException
+     *     If any error at all occurs during decryption.
+     */
+    public byte[] decrypt(Key key, byte[] cipherText) throws GuacamoleException {
+
+        try {
+
+            // Init cipher for descryption using secret key
+            Cipher cipher = Cipher.getInstance(DECRYPTION_CIPHER_NAME);
+            cipher.init(Cipher.DECRYPT_MODE, key, NULL_IV);
+
+            // Perform decryption
+            return cipher.doFinal(cipherText);
+
+        }
+
+        // Rethrow all decryption failures identically
+        catch (InvalidAlgorithmParameterException e) {
+            throw new GuacamoleServerException(e);
+        }
+        catch (NoSuchAlgorithmException e) {
+            throw new GuacamoleServerException(e);
+        }
+        catch (NoSuchPaddingException e) {
+            throw new GuacamoleServerException(e);
+        }
+        catch (InvalidKeyException e) {
+            throw new GuacamoleServerException(e);
+        }
+        catch (IllegalBlockSizeException e) {
+            throw new GuacamoleServerException(e);
+        }
+        catch (BadPaddingException e) {
+            throw new GuacamoleServerException(e);
+        }
```

Comment:
Any reason not to use the multi-catch, here?

Reply:
Sure - I'll switch instances of this over to multi-catch.